neo/Ledger/MemoryPool.cs

```diff
@@ -4,162 +4,376 @@
+        {
+            verifiedTxRwLock.EnterWriteLock();
+            try
+            {
+                // First remove the transactions verified in the block.
+                foreach (Transaction tx in block.Transactions)
                 {
-                    RemoveLowestFee(_mem_pool_free, exceed);
-                    exceed = Count - Capacity;
-
-                    if (exceed > 0)
+                    if (!TryRemoveInternal(tx.Hash, out _))
                     {
-                        RemoveLowestFee(_mem_pool_fee, exceed);
+                        if (_unverifiedTransactions.TryRemove(tx.Hash, out PoolItem item))
+                        {
+                            SortedSet<PoolItem> pool = tx.IsLowPriority
+                                ? _unverifiedSortedLowPriorityTransactions
+                                : _unverifiedSortedHighPriorityTransactions;
+                            pool.Remove(item);
+                        }
                     }
                 }
-            }
 
-            return pool.ContainsKey(hash);
+                // Add all the previously verified transactions back to the unverified transactions
+                foreach (PoolItem item in _sortedHighPrioTransactions)
+                {
+                    _unverifiedTransactions.TryAdd(item.Transaction.Hash, item);
+                    _unverifiedSortedHighPriorityTransactions.Add(item);
+                }
+
+                // NOTE: This really shouldn't be necessary any more and can actually be counter-productive, since
+                //       they can be re-added anyway and would incur a potential addtional verification; setting to 1000
+                //       blocks for now. If transactions want to only be valid for short times they can include a script.
+                var lowPriorityCutOffTime = DateTime.UtcNow.AddSeconds(-Blockchain.SecondsPerBlock * 1000);
+                foreach (PoolItem item in _sortedLowPrioTransactions)
+                {
+                    // Expire old free transactions
+                    if (item.Transaction.IsLowPriority && item.Timestamp < lowPriorityCutOffTime) continue;
+
+                    _unverifiedTransactions.TryAdd(item.Transaction.Hash, item);
+                    _unverifiedSortedLowPriorityTransactions.Add(item);
+                }
+
+                // Clear the verified transactions now, since they all must be reverified.
+                _unsortedTransactions.Clear();
+                _sortedHighPrioTransactions.Clear();
+                _sortedLowPrioTransactions.Clear();
+
+                // If we know about headers of future blocks, no point in verifying transactions from the unverified tx pool
+                // until we get caught up.
+                if (block.Index < Blockchain.Singleton.HeaderHeight)
+                    return;
+
+                ReverifyTransactions(_sortedHighPrioTransactions, _unverifiedSortedHighPriorityTransactions,
```

Comment:
If the node is being bombarded with new transactions constantly that are low priority it will prevent Idle from running much and re-verifying the highest fee transactions. In order to ensure high fee transactions get in a block first it is important to immediately give some time to reverify them right after the block persists.